src/ConstructiveSolidGeometry/PointsAndVectors.jl

```diff
@@ -22,36 +22,30 @@ end
 
 @inline _in_φ(p::CartesianPoint{T}, φ::AbstractInterval) where {T} = mod(atan(p.y, p.x), T(2π)) in φ    
 
-@inline _in_z(p::CartesianPoint, z::Real) = abs(p.z) <= z
-@inline _in_z(p::CartesianPoint, z::AbstractInterval) = p.z in z
+@inline _in_x(p::CartesianPoint, x::Real) = abs(p.x) <= x
+@inline _in_x(p::CartesianPoint, x::AbstractInterval) = p.x in x
 
-# have to add _in_x & _in_y
+@inline _in_y(p::CartesianPoint, y::Real) = abs(p.y) <= y
+@inline _in_y(p::CartesianPoint, y::AbstractInterval) = p.y in y
 
-"""
-    struct CartesianVector{T} <: AbstractCoordinateVector{T, Cartesian}
+@inline _in_z(p::CartesianPoint, z::Real) = abs(p.z) <= z
+@inline _in_z(p::CartesianPoint, z::AbstractInterval) = p.z in z
 
-* `x`: x-component in meter
-* `y`: y-component in meter
-* `z`: z-component in meter
-"""
-struct CartesianVector{T} <: AbstractCoordinateVector{T, Cartesian}
-    x::T
-    y::T
-    z::T
-end
+@inline _in_radius(p::CartesianPoint, radius::Real) = norm(p) <= radius
+@inline _in_radius(p::CartesianPoint, radius::AbstractInterval) = norm(p) in radius
```

Comment:
Naming is not clear. `radius` can be seen different for polar and spherical coordinates. I guess this is for the cylindrical case, and, thus, `norm(p)` does unnecessary calculations

Reply:
I know I defined `_in_r` but that just crossed my mind.

Reply:
Ah, I see. `_in_r` is for cylindrical and `_in_radius` for spherical coordinates?
We should change this to `_in_r(p, z, ::Type{Cylindrical})` and add `abstract type Spherical`.

Reply:
We could also go for the convention to use `ρ` for radii in cylindrical coordinates and `r` for radii in spherical coordinates.
(This is the general convention, see the Wikipedia entries for [cylindrical](https://en.wikipedia.org/wiki/Cylindrical_coordinate_system) and [spherical](https://en.wikipedia.org/wiki/Spherical_coordinate_system) coordinates).
However, `r` has been used throughout the whole SSD code and might cause confusions with the impurity charge density.

Reply:
Hmm, I don't know how known this convention is. I would rather go for adding an additional argument. Or two functions with different naming `_in_cyl_r` / `_in_sph_r` or so.

Reply:
Ah, and please use `hypot(p[1], p[2], p[3])` for the spherical radius